wazp/callbacks.py

```diff
@@ -1,58 +1,113 @@
 import base64
 import pathlib as pl
 import re
+from typing import Any
 
 import dash
 import dash_bootstrap_components as dbc
 import utils
 import yaml
-from dash import Input, Output, State, html
+from dash import Input, Output, State, dash_table, html
 
 VIDEO_TYPES = [".avi", ".mp4"]
 # TODO: other video extensions? have this in project config file instead?
 
 
-def get_metadata_callbacks(app: dash.Dash) -> None:
+def get_home_callbacks(app: dash.Dash) -> None:
     """
-    Return all metadata callback functions
+    Return all home callback functions
 
     """
 
     @app.callback(
-        Output("output-data-upload", "children"),
+        Output("session-storage", "data"),
+        Output("upload-message", "is_open"),
+        Output("upload-message", "children"),
+        Output("upload-message", "color"),
         Input("upload-data", "contents"),
         State("upload-data", "filename"),
+        State("upload-message", "is_open"),
     )
-    def update_file_drop_output(up_content: str, up_filename: str) -> html.Div:
+    def save_input_config_to_storage(
+        up_content: str, up_filename: str, up_message_state: bool
+    ) -> tuple[tuple[Any, Any], bool, str, str]:
         """
-        Read uploaded config file and return component with:
-        - table with metadata per video,
-        - auxiliary buttons for common table manipulations
+        Save input config to temp shared memory
+        see https://community.plotly.com/t/
+        dash-plotly-share-callback-input-in-another-page
+        -with-dcc-store/44190/2
 
         """
-        # Read uploaded content
+        data_to_store = ((), ())
+        output_message = ""
+        output_color = "light"
         if up_content is not None:
             _, content_str = up_content.split(",")
             try:
                 if "yaml" in up_filename:
                     # get config
                     cfg = yaml.safe_load(base64.b64decode(content_str))
-                    # get video dir
-                    video_dir = cfg["videos_dir_path"]
+
                     # get metadata fields dict
                     with open(cfg["metadata_fields_file_path"]) as mdf:
                         metadata_fields_dict = yaml.safe_load(mdf)
+
+                    # bundle data
+                    data_to_store = (cfg, metadata_fields_dict)
```

Comment:
Should we store this as a dict("config"=cfg, "metadata_fields"=metadata_fields_dict) instead? Or perhaps as a [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple)? So we don't have to remember what is stored in which position. I like the dictionary solution, since we can add more things to session-storage as needed, each with their own keys.